src/lib/translators/mongo.ts

```diff
@@ -527,6 +527,91 @@ function transformSplit(step: Readonly<S.SplitStep>): MongoStep {
+  // Some useful formula:
+  const quintilesIndexFormula: any = {
+    maxIndex: { $subtract: ['$count', 1] },
+    lastQuintileIndex: { $multiply: [{ $divide: ['$count', 10] }, 9] },
+    lastQuartileIndex: { $multiply: [{ $divide: ['$count', 4] }, 3] },
+    medianIndex: { $divide: ['$count', 2] },
+    firstQuartileIndex: { $divide: ['$count', 4] },
+    firstQuintileIndex: { $divide: ['$count', 10] },
+    minIndex: 0,
+  };
+  const varianceFormula: any = { $subtract: ['$average_sum_square', { $pow: ['$average', 2] }] };
+  const otherStatisticsFormula: any = {
+    count: 1,
+    average: 1,
+    variance: varianceFormula,
+    std: {
+      // std = square_root(N/(N-1) * variance) (With N = count) to get the unbiased version of std
+      $pow: [
+        { $multiply: [{ $divide: ['$count', { $subtract: ['$count', 1] }] }, varianceFormula] },
+        0.5,
+      ],
+    },
+  };
+  const isQuintileStatistics = (statistic: string): boolean =>
+    Object.keys(quintilesIndexFormula).includes(`${statistic}Index`);
+  // ---
+  // Some explicit conditions to compute average or columns square:
+  const computeColumnSquare = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('variance') || step.statistics.includes('std');
+  const computeAverage = (step: Readonly<S.StatisticsStep>): boolean =>
+    step.statistics.includes('average') ||
+    step.statistics.includes('variance') ||
+    step.statistics.includes('std');
+  // ---
+
+  const stepQuantileIndex: any = { $addFields: {} };
+  let statistic: string;
+  for (statistic of step.statistics) {
+    if (isQuintileStatistics(statistic)) {
+      stepQuantileIndex.$addFields[`${statistic}Index`] = {
+        $trunc: quintilesIndexFormula[`${statistic}Index`], //TODO: it would be more canonical to use a round version of index here. Migrating to Mongo 4.2 will allow use to use the $round step.
+      };
+    }
+  }
+  const stepProjectStatistics: any = { $project: {} };
+  for (statistic of step.statistics) {
+    if (isQuintileStatistics(statistic)) {
+      stepProjectStatistics.$project[statistic] = {
+        $arrayElemAt: ['$data', `$${statistic}Index`],
+      };
+    } else {
+      stepProjectStatistics.$project[statistic] = otherStatisticsFormula[statistic];
+    }
+  }
+
+  return [
+    {
+      $project: {
+        column: `$${[step.column]}`,
+        column_square: computeColumnSquare(step) ? { $pow: [`$${[step.column]}`, 2] } : undefined,
+      },
+    },
+    {
+      $sort: { column: 1 },
+    },
+    {
+      $group: {
+        _id: 'whatever',
+        count: { $sum: 1 },
+        average_sum_square: computeColumnSquare(step) ? { $avg: '$column_square' } : undefined, // I am using this formula of the variance: avg(x^2) - avg(x)^2
+        average: computeAverage(step) ? { $avg: '$column' } : undefined, // No need to compute average if neither average or variance are requested
+        data: { $push: '$column' },
```

Comment:
Regarding the management of null values, a solution I suggest is to add an `$addField` stage before the next step to project a `dataNoNull` using the [`$filter`](https://docs.mongodb.com/manual/reference/operator/aggregation/filter/) operator, and to use this field as a basis for the quantiles computation